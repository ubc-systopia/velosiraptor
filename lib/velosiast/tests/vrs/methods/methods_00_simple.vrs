
// method returning a bool
fn foo() -> bool {
    true
}

// method returning an int
fn bar() -> int {
    3
}

// method with one argument
fn baz(v: bool) -> bool {
    !v
}

// method returning an int
fn bar(v: bool, u: bool) -> bool {
    v && u
}

// method returning an int
fn bar(v: int, u: int) -> int {
    v & u
}

// parameter names
fn bar_bar(v_3: int, u: int) -> int {
    v_3 & u
}