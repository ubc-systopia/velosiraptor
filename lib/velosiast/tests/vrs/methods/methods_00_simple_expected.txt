  fn foo() -> bool
    requires true;
  {
    true
  }

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

  fn bar() -> int
    requires true;
  {
    0x3
  }

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

  fn baz(v: bool) -> bool
    requires true;
  {
    !(v)
  }

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

  fn bar(v: bool, u: bool) -> bool
    requires true;
  {
    v && u
  }

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

  fn bar(v: int, u: int) -> int
    requires true;
  {
    v & u
  }

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

  fn bar_bar(v_3: int, u: int) -> int
    requires true;
  {
    v_3 & u
  }

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

